test/whitelisted_crowdsale.js

```diff
@@ -53,22 +46,62 @@ contract('WhitelistedCrowdsale', async (accounts) => {
+      await instanceWhitelistedCrowdsale.rate.call(), defaultRateForPrivateOffer
     );
     assert.equal(
-      await instanceWhitelistedCrowdsale.minPurchase.call(), minPurchase
+      await instanceWhitelistedCrowdsale.minPurchase.call(), defaultMinPurchaseForPrivateOffer
     );
     assert.equal(
-      await instanceWhitelistedCrowdsale.maxPurchase.call(), maxPurchase
+      await instanceWhitelistedCrowdsale.maxPurchase.call(), defaultMaxPurchaseForPrivateOffer
     );
     assert.equal(
-      await instanceWhitelistedCrowdsale.wallet.call(), wallet
+      await instanceWhitelistedCrowdsale.wallet.call(), escrowAccount
     );
     assert.equal(
       await instanceWhitelistedCrowdsale.token.call(), token.address
     );
     assert.equal(
-      await instanceWhitelistedCrowdsale.countPurchaseAmount.call(), countPurchaseAmount
+      await instanceWhitelistedCrowdsale.countPurchaseAmount.call(), defaultCountPurchaseAmountForPrivateOffer
+    );
+  });
+
+  it("should throw error when the beneficiary not in whitelist", async () => {
+    try {
+      await instanceWhitelistedCrowdsale.buyTokens.sendTransaction(userNotFromWhitelist, {from: owner, value: web3.toWei(2, "ether")});
+      assert.ifError('Error, the owner should not be able to call this method');
+    } catch (err) {
+      assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Benefeciary not in whitelist trying to buy tokens");
+    }
+  });
+
+  it("should throw error when the beneficiary has a zero address", async () => {
+    try {
+      await instanceWhitelistedCrowdsale.buyTokens.sendTransaction(0, {from: owner, value: web3.toWei(2, "ether")});
+      assert.ifError('Error, the owner should not be able to call this method');
+    } catch (err) {
+      assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Benefeciary has zero address");
+    }
+  });
+
+  it("should throw error when tokens not enough", async () => {
+    let tokenBalance = await token.balanceOf(instanceWhitelistedCrowdsale.address);
+    let weiAmount = tokenBalance.div(await instanceWhitelistedCrowdsale.rate.call()).add(BigNumber('10'));
+
+    try {
+      await instanceWhitelistedCrowdsale.buyTokens.sendTransaction(userFromWhitelist, {from: owner, value: weiAmount});
+      assert.ifError('Error, the owner should not be able to call this method');
+    } catch (err) {
+      assert.equal(err, 'Error: VM Exception while processing transaction: revert', "Tokens not enough");
+    }
+  });
+
+  it("should forward funds when tokens were bought", async () => {
+    let weiAmount = web3.toWei(2, "ether");
+    let walletAddress = await instanceWhitelistedCrowdsale.wallet.call();
+    let walletBalance = web3.eth.getBalance(walletAddress);
+    await instanceWhitelistedCrowdsale.buyTokens.sendTransaction(userFromWhitelist, {from: owner, value: weiAmount});
+    assert.isOk(
+      walletBalance.add(weiAmount).eq(web3.eth.getBalance(walletAddress))
```

Comment:
I think we should also check that the user has received the expected amount of tokens.